### ISSUE TITLE
Layer Offset may not be working correctly

### ISSUE THREAD

**Issue by wollyroche** [joshuajbouw/bevy_tilemap#156]
**Bevy Tilemap version**

0.4.0

**Operating system & version**

Windows 10

**What you did**

Tried to offset a tile layer to make tiles appear 3D using z_layers, add_layer, and layer_offset.

**What you expected to happen**

Two things:

1. Originally I thought that each subsequent tile layer after the first would be offset by the amount specified in layer_offset, but then I thought ...
2. Wouldn't the layer_offset just offset all layers as it is only a single value on the Tilemap, therefore the whole map would be offset and not individual layers? - is there a way to offset a single layer? Like on the add_layer function?

**What actually happened**

Nothing was offset.

![image](https://user-images.githubusercontent.com/29350785/115096899-5f15f380-9f7b-11eb-82e7-e04132193843.png)

**Additional information**

This was tested on an alteration of your Hex Tile Odd Rows example.

The only alterations being that I changed the hex to a 3d one and used just a single chunk.

As well as the below code.

![image](https://user-images.githubusercontent.com/29350785/115096909-6c32e280-9f7b-11eb-8194-c26ccfbb18f7.png)

![image](https://user-images.githubusercontent.com/29350785/115096924-78b73b00-9f7b-11eb-88a3-ffb890dff600.png)

I'm sorry if I'm getting something obviously wrong here, I've tried reading the doc comments on the function calls but I couldn't find any examples in the repo that used layer_offset.

Thanks for any help you might have,  you've created an excellent extensible plugin here.

**Comment by joshuajbouw** [comment]
Yeah, once 0.5 hit, lots of stuff was rushed out to get the release in on time. I hadn't made any examples to test all the new features yet.

**Comment by wollyroche** [comment]
No problem - just an update that's a little more easier to follow than my custom code above.

I edited the random_dungeon example just to include the call to layer_offset.

![image](https://user-images.githubusercontent.com/29350785/115125889-524adb80-a01f-11eb-98ee-26efe6ffdd84.png)

Which also doesn't offset any layers.

![image](https://user-images.githubusercontent.com/29350785/115125912-81614d00-a01f-11eb-9931-5ef6de527d84.png)

Thanks for the reply and the great work

**Comment by wollyroche** [comment]
I've tried to do some debugging on where the z_offset is used in creating the vertices.

The calculation looks like its working fine, the number of indices is also correct.

Tests done on the hex_tile_odd_rows example.

![image](https://user-images.githubusercontent.com/29350785/115127280-6bf12080-a029-11eb-81f6-5928ac7c92ce.png)

![image](https://user-images.githubusercontent.com/29350785/115127290-81fee100-a029-11eb-9945-e4d06832c6b8.png)

![image](https://user-images.githubusercontent.com/29350785/115127319-b377ac80-a029-11eb-8d12-518ae54fbd3c.png)

But it will still only display 1 depth of tiles, not the second offset tiles.

![image](https://user-images.githubusercontent.com/29350785/115127325-bd011480-a029-11eb-8bbd-ca6a41ba1aa0.png)

**Comment by wollyroche** [comment]
Did some more testing.

I thought I might have found it as I wasn't adding the z index to the Tile points, which I tried doing.

![image](https://user-images.githubusercontent.com/29350785/115128080-6dbde280-a02f-11eb-96af-2d046bc160dc.png)

I was sure this would fix it but it still isn't displaying, but I am getting a warning about a tile being out of bounds - even with enough layers added.

![image](https://user-images.githubusercontent.com/29350785/115128121-a362cb80-a02f-11eb-9d2a-9e4a35b51b62.png)

![image](https://user-images.githubusercontent.com/29350785/115128107-9940cd00-a02f-11eb-9893-d4d5bdd2ca68.png)

**Comment by joshuajbouw** [comment]
Hmm, thanks for this. I'm quite a bit behind on my game project and it is a requirement for it. I hope to have it resolved shortly.
